### PR TITLE
mpv 0.30

### PR DESCRIPTION
**Pull request by jnozsc**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by jnozsc** [comment]
I don't know why the @BrewTestBot suggest `uses_from_macos: libarchive`, but when I test on commit https://github.com/Homebrew/homebrew-core/pull/45854/commits/0ddf90a1c34fa97dcff145f0e1c332ee57149131   

I get error message as 
```
Checking for libarchive wrapper for reading zip files and more            : no ('libarchive >= 3.0.0' not found) 
You manually enabled the feature 'libarchive', but the autodetection check failed.
```

I check the code from mpv https://github.com/mpv-player/mpv/blame/master/wscript#L416 , and I don't think mpv actually looking for the one @BrewTestBot  or  `uses_from_macos` points to.

is there any workaround to skip this warning or fix it? Thanks

**Comment by chenrui333** [comment]
Looks like we have removed it per this PR, https://github.com/Homebrew/homebrew-core/pull/44282?

**Comment by jnozsc** [comment]
yes, I believe `mpv` 0.29 does not support latest version of `ffmpeg`, which block the unit test for the PR to upgrade `ffmpeg` to 4.2.1, and it seems back to that time, before Catalina releases, mpv team do not think it is necessary to release a new version for `mpv` 

but now, the `mpv` 0.30 fix the issue

File: Formula/mpv.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  depends_on "pkg-config" => :build
+  depends_on "python" => :build
+
+  depends_on "ffmpeg" => "4.0"
+  depends_on "jpeg"
+  # uses_from_macos "libarchive" does not work
+  depends_on "libarchive"
+  depends_on "libass"
+  depends_on "little-cms2"
+  depends_on "lua@5.1"
```
Comment (fxcoudert) [inline Formula/mpv.rb]: It doesn't build with latest lua version (5.3)?

File: Formula/mpv.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  homepage "https://mpv.io"
+  url "https://github.com/mpv-player/mpv/archive/v0.30.0.tar.gz"
+  sha256 "33a1bcb7e74ff17f070e754c15c52228cf44f2cefbfd8f34886ae81df214ca35"
+  head "https://github.com/mpv-player/mpv.git"
+
+  depends_on "docutils" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python" => :build
+
+  depends_on "ffmpeg" => "4.0"
```
[the remark this answers is not in the document]
Comment (fxcoudert) [reply Formula/mpv.rb]: We don't accept pinned versions of dependencies. Dependencies are “latest version”, always.

File: Formula/mpv.rb (hunk 1 above)
Comment (jnozsc) [reply Formula/mpv.rb]: unfortunately, mpv does not support lua 5.3 at this point https://github.com/mpv-player/mpv/issues/5205

File: Formula/mpv.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jnozsc) [reply Formula/mpv.rb]: fix in https://github.com/Homebrew/homebrew-core/pull/45854/commits/589c060ae7a5edfdb3ebca5e26e49aa3f8683743

File: Formula/mpv.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  depends_on "jpeg"
+  # uses_from_macos "libarchive" does not work
```
[the remark this answers is not in the document]
Comment (jnozsc) [reply Formula/mpv.rb]: fix in https://github.com/Homebrew/homebrew-core/pull/45854/commits/589c060ae7a5edfdb3ebca5e26e49aa3f8683743

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@Homebrew/core opinion on this?

**Comment by SMillerDev** [comment]
Seeing how bad the experience packaging mpv has been I'm inclined to say it should just be in a tap somewhere.

File: Formula/mpv.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      --docdir=#{doc}
+      --zshdir=#{zsh_completion}
+    ]
+
+    system "./bootstrap.py"
+    system "python3", "waf", "configure", *args
+    system "python3", "waf", "install"
+
+    system "python3", "TOOLS/osxbundle.py", "build/mpv"
+    prefix.install "build/mpv.app"
```
Comment (MikeMcQuaid) [inline Formula/mpv.rb]: Don't install a `.app` for Homebrew/homebrew-core, thanks. That's what `homebrew/cask` is for.

**Comment by MikeMcQuaid** [comment]
Once the `.app` is removed: I'm fine with including this. Ultimately if our users want it: we may as well provide it.

File: Formula/mpv.rb (hunk 4 above)
Comment (jnozsc) [reply Formula/mpv.rb]: thank @MikeMcQuaid , it seems that install as a `.app` was an option, then it became default behavior in this commit https://github.com/Homebrew/homebrew-core/commit/1fdd9b152760e9d743964dd955c1ae816a3ca191  do you want me to add the `with-bundle` option back?

Comment (fxcoudert) [reply Formula/mpv.rb]: No option, simply do not install the app. People who want the app can install from cask

Comment (jnozsc) [reply Formula/mpv.rb]: deleted in https://github.com/Homebrew/homebrew-core/pull/45854/commits/0c7e9af5fd5b41024dba277a5eef59c974168fa5

**Comment by fxcoudert** [comment]
OK, we're good to go. Please squash all changes into a single commit (`mpv 0.30`) and we can merge!

**Comment by MikeMcQuaid** [comment]
Thanks so much again for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @jnozsc!

**Pull request closed without merging**